modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/provider/ObjectEntryInfoItemFieldValuesProvider.java

```diff
@@ -389,31 +408,75 @@ private List<InfoFieldValue<Object>> _getInfoFieldValues(
 			_objectFieldLocalService.getObjectFields(
 				serviceBuilderObjectEntry.getObjectDefinitionId());
 
-		objectEntryFieldValues.addAll(
-			TransformUtil.transform(
-				objectFields,
-				objectField -> new InfoFieldValue<>(
-					InfoField.builder(
-					).infoFieldType(
-						ObjectFieldDBTypeUtil.getInfoFieldType(objectField)
-					).namespace(
-						ObjectField.class.getSimpleName()
-					).name(
-						objectField.getName()
-					).labelInfoLocalizedValue(
-						InfoLocalizedValue.<String>builder(
-						).values(
-							objectField.getLabelMap()
-						).build()
-					).build(),
-					_getValue(objectField, properties))));
+		for (ObjectField objectField : objectFields) {
+			Object value = properties.get(objectField.getName());
+
+			objectEntryFieldValues.add(_getInfoFieldValue(objectField, value));
+
+			List<InfoFieldValue<Object>> relatedObjectInfoFieldValues =
+				_getRelatedObjectEntryFieldValues(objectField, value);
+
+			if (ListUtil.isNotEmpty(relatedObjectInfoFieldValues)) {
+				objectEntryFieldValues.addAll(relatedObjectInfoFieldValues);
+			}
+		}
```

Comment:
@jkappler 

I believe we can create a private method in this class to put this snippet of code to avoid duplication.

Reply:
changed @MarinhoFeliphe